docs/src/modules/components/AppTableOfContents.js

```diff
@@ -201,83 +176,107 @@ class AppTableOfContents extends React.Component {
     }
   };
 
-  handleClick = hash => () => {
+  // Update the active TOC entry if the hash changes through click on '#' icon
+  const handleHashChange = () => {
+    const hash = window.location.hash.substring(1);
+
+    if (activeState !== hash) {
+      setActiveState(hash);
+    }
+  };
+
+  const handleScroll = throttle(() => {
+    findActiveIndex();
+  }, 166); // Corresponds to 10 frames at 60 Hz.
+
+  React.useEffect(() => {
+    itemsClient = getItemsClient(itemsServer);
+
+    window.addEventListener('hashchange', handleHashChange);
+
+    return function componentWillUnmount() {
```

Comment:
That is not correct. You didn't specify dependencies in the second argument to `useEffect`. The effect will be executed after every render (as well as this cleanup).

Reply:
Not incorrect, just not optimal (but I had already acknowledge that). Thanks for the pointer - I guess I should have R'dTFM!

Reply:
No it's simply not correct. That function will execute not only on unmount but also before every other effect execution. In this case the effect will scheduled after every render. This is similar to componentDidUpdate +  componentWillUnmount. 

Doesn't matter if it is not optimal. You either change the effect hook so that it only runs on mount or you change the name of the cleanup function.

Reply:
> No it's simply not correct.
[...]
> change the name of the cleanup function

Oh, the function name! Why didn't you just say so? It might as well be called Bob, (or be anonymous) for all I care. The name (if it is going to have one) is correct for its intended purpose, just not the way it's currently called (in the current state of the PR, as opposed to my local branch).

Reply:
> Oh, the function name! Why didn't you just say so?

Oh no I did the very same thing I asked you to earlier: Provide context. Sorry. Yes this is specifically about the name. From the cleanup name it looks like you missed the second argument for useEffect. If the second argument is intentionally missing then the cleanup function name is misleading. Only one of those statements should be true.

 This
`React.useEffect(() => function componentWillUnmount() {})` is similar to

```
// class body
componentWillUnmount() {}
componentDidUpdate = this.componentWillUnmount;
```